src/psyclone/tests/psyir/transformations/replace_reference_by_literal_trans_test.py

```diff
@@ -182,101 +178,179 @@ def test_rrbl_array_shape(fortran_reader, fortran_writer):
     assert "if (.false.) then" in written_code
 
 
-def test_array_type_extend(fortran_reader, fortran_writer):
+def test_rrbl_array_type_extent(fortran_reader, fortran_writer):
+    """test replacement of lower bound of an array dimension"""
+
     source = """subroutine foo()
     integer, parameter ::  a = 3
     integer, dimension(:,a:) :: x
     end subroutine"""
     psyir = fortran_reader.psyir_from_source(source)
-    foo: Routine = psyir.walk(Routine)[0]
+    routine_foo: Routine = psyir.walk(Routine)[0]
     rbbl = ReplaceReferenceByLiteralTrans()
-    rbbl.apply(foo)
-    written_code = fortran_writer(foo)
+    rbbl.apply(routine_foo)
+    written_code = fortran_writer(routine_foo)
 
     assert "integer, dimension(:,3:)" in written_code
 
 
-def test_raise_transformation_error_symbol_table_is_none(
+def test_rrbl_conflict_between_module_and_routine_scopes(
     fortran_reader, fortran_writer
 ):
-    source = """subroutine foo()
-    integer, parameter ::  a = 3
-    integer :: x
-    x = a
-    end subroutine"""
+    """test fortran code annotation with transformation warning"""
+
+    source = """module toto
+    integer, parameter :: a=3
+contains
+    subroutine tata()
+    call foo()
+    end subroutine
+    subroutine foo()
+        integer, parameter :: x=a
+        integer, parameter :: a=2
+        integer, parameter :: y=a
+        end subroutine
+        end module
+        """
+    psyir = fortran_reader.psyir_from_source(source)
+    routine_foo: Routine = psyir.walk(Routine)[1]
+    rbbl = ReplaceReferenceByLiteralTrans()
+    rbbl.apply(routine_foo)
+    written_code = fortran_writer(routine_foo.ancestor(Container))
+    print(written_code)
+    assert (
+        "! Psyclone(ReplaceReferenceByLiteralTrans): Symbol already found a"
+        in written_code
+    )
+    assert "integer, parameter :: x = a" in written_code
+    assert "integer, parameter :: a = 2" in written_code
+    assert "integer, parameter :: y = a" in written_code
+
+
+def test_rrbl_call_with_const(fortran_reader, fortran_writer):
+    """
+    * constant var as argument to subroutine call, e.g. call my_sub(ic1+u3)
+    """
+
+    source = """module toto
+    integer, parameter :: a=3, u2=2
+contains
+    subroutine tata()
+    integer :: i
+    integer, parameter :: u3=3
+    do i=1,10,1
+        call foo(u2+ u3)
+    enddo
+    end subroutine
+    subroutine foo()
+        integer, dimension(10, a) :: array
+        end subroutine
+        end module
+        """
+    psyir = fortran_reader.psyir_from_source(source)
+    routine_foo: Routine = psyir.walk(Routine)[0]
```

Comment:
Isn't this getting the 'tata' routine? If so, please rename the variable to avoid confusion.